webserver/views/api.py

```diff
@@ -205,6 +205,13 @@ def get_messybrainz_data(listen):
 
 def _validate_listen(listen):
     """Make sure that required keys are present, filled out and not too large."""
+
+    if not 'listened_at' in listen:
+        _log_raise_400("JSON document must contain the key listened_at at the top level.", listen)
+        
+    if type(listen['listened_at']) != int:
```

Comment:
do we want this beahaviour? an alternative would be 
```
try:
    int(listen['listened_at'])
except ValueError:
    raise here
```
which would also let us accept strings from less-robust clients.